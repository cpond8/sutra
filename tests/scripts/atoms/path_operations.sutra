;; path_operations.sutra
;; Test user-facing path operations: set!, get, del!

;; Basic path operations
(set! name "Alice")
(print (get name))

(set! age 25)
(print (get age))

;; Nested path operations
(set! person.name "Bob")
(set! person.age 30)
(print (get person.name))
(print (get person.age))

;; List path operations
(set! items.0 "first")
(set! items.1 "second")
(print (get items.0))
(print (get items.1))

;; Delete operations
(del! age)
(print (get age))

;; Test getting non-existent keys
(print (get nonexistent))
