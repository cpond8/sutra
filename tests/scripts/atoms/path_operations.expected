Alice
25
Bob
30
first
second
nil
nil
